mypyc/ops.py

```diff
@@ -391,22 +405,56 @@ def to_lines(self) -> List[str]:
 ERR_FALSE = 2  # Generates false (bool) on exception
 
 
-class Op:
+class Value:
     # Source line number
     line = -1
+    name = '?'
+    type = void_rtype  # type: RType
 
     def __init__(self, line: int) -> None:
         self.line = line
 
-    def can_raise(self) -> bool:
-        # Override this is if Op may raise an exception. Note that currently the fact that
-        # only RegisterOps may raise an exception in hard coded in some places.
-        return False
+    @property
+    def is_void(self) -> bool:
+        return isinstance(self.type, RVoid)
 
     @abstractmethod
     def to_str(self, env: Environment) -> str:
         raise NotImplementedError
 
+
+class Register(Value):
+    def __init__(self, type: RType, line: int = -1) -> None:
+        super().__init__(line)
+        self.type = type
+
+    def to_str(self, env: Environment) -> str:
+        return self.name
+
+    @property
+    def is_void(self) -> bool:
+        return False
+
+
+# Unfortunately we have visitors which are statement-like rather than expression-like.
+# It doesn't make sense to have the visitor return Optional[Value] because every
+# method either always returns no value or returns a value.
+#
+# Eventually we may want to separate expression visitors and statement-like visitors at
+# the type level but until then returning INVALID_VALUE from a statement-like visitor
+# seems acceptable.
+INVALID_VALUE = Register(void_rtype)
```

Comment:
What about giving this a name that makes it obvious what is happening if this is accidentally being used as a value?